ros_translator/python/ros_translator/core.py

```diff
@@ -0,0 +1,89 @@
+import rospy
+from std_msgs.msg import String
+from deep_translator import (
+    GoogleTranslator,
+    MicrosoftTranslator,
+    PonsTranslator,
+    LingueeTranslator,
+    MyMemoryTranslator,
+    YandexTranslator,
+    PapagoTranslator,
+    #DeeplTranslator,
+    #QcriTranslator,
+    single_detection,
+    batch_detection)
+from dynamic_reconfigure.server import Server
+from ros_translator.cfg import ROSTranslatorConfig
+
+engines = {
+    'google': GoogleTranslator,
+    'microsoft': MicrosoftTranslator,
+    'pons': PonsTranslator,
+    'linguee': LingueeTranslator,
+    'mymemory': MyMemoryTranslator,
+    'yandex': YandexTranslator,
+    'papago': PapagoTranslator,
+    #'deepl': DeeplTranslator,
+    #'qcri': QcriTranslator,
+}
+
+
+class ROSTranslator(object):
+
+  def __init__(self):
+
+    self.from_language = rospy.get_param('~from_language', 'auto')
+    self.to_language = rospy.get_param('~to_language', 'en')
+    self.translator = rospy.get_param('~translator', 'google')
+    self.api_key = rospy.get_param('~api_key', '')
+    self.engine = None
+    self.update_engine()
+
+    self.param_srv = Server(ROSTranslatorConfig, self.param_callback)
+    self.pub = rospy.Publisher('~output_text', String, queue_size=1)
+    self.sub = rospy.Subscriber('~input_text', String, self.callback)
+
```

Comment:
How about adding actionlib server?

Reply:
I don't think actionlib server is not necessary since if we just want to translate some text in one node. It is enough to use `deep-translator` library directory.